autosklearn/evaluation/train_evaluator.py

```diff
@@ -922,15 +922,15 @@ def get_splitter(self, D):
                         raise ValueError('Must provide parameter groups '
                                          'for chosen CrossValidator.')
                     try:
-                        if self.resampling_strategy_args['groups'].shape != y.shape:
+                        if self.resampling_strategy_args['groups'].shape[-1] != y.shape[-1]:
```

Comment:
Could you please let me know when this fails? Also, could you please update the error message?

Reply:
With removing the ravel above, shape[-1] represents the number of samples whatever type of target it is. It fail when there is a for example multi-output regression target with n_samples = 100, n_targets = 3.  If we keep the ravel in line 912 and keep line 925 as shape. The 'groups' we have to define is (300,) in shape. There, however are only 100 samples which led to this error message. 

```
[ERROR] [2020-07-22 22:27:55,917:AutoML(1):d4069f363ae29ee607f565032a35aa5f] Error creating dummy predictions: 
{'traceback': 'Traceback (most recent call last):\n  File "/home/charles/anaconda3/envs/autoskdev/lib/python3.7/site-
packages/autosklearn/evaluation/__init__.py", line 29, in fit_predict_try_except_decorator\n    return ta(queue=queue,
 **kwargs)\n  File "/home/charles/anaconda3/envs/autoskdev/lib/python3.7/site-
packages/autosklearn/evaluation/train_evaluator.py", line 1239, in eval_cv\n    
.fit_predict_and_loss(iterative=iterative)\n  File "/home/charles/anaconda3/envs/autoskdev/lib/python3.7/site-
packages/autosklearn/evaluation/train_evaluator.py", line 446, in fit_predict_and_loss\n    
groups=self.resampling_strategy_args.get(\'groups\')\n  File "/home/charles/anaconda3/envs/autoskdev/lib/python3.7/site-
packages/sklearn/model_selection/_split.py", line 327, in split\n    X, y, groups = indexable(X, y, groups)\n  File 
"/home/charles/anaconda3/envs/autoskdev/lib/python3.7/site-packages/sklearn/utils/validation.py", line 248, in indexable\n    
check_consistent_length(*result)\n  File "/home/charles/anaconda3/envs/autoskdev/lib/python3.7/site
-packages/sklearn/utils/validation.py", line 212, in check_consistent_length\n    " samples: %r" % [int(l) for l in 
lengths])\nValueError: Found input variables with inconsistent numbers of samples: [100, 100, 300]\n', 'error': 
"ValueError('Found input variables with inconsistent numbers of samples: [100, 100, 300]')", 'configuration_origin': 
'DUMMY'} 
```

I have tried it manually here and a clearer version is shown in the photo.

![groups_error](https://user-images.githubusercontent.com/14807898/88227543-0f193300-cc6e-11ea-88c7-a0c498cdcbb9.png)